server/core/include/process_manager.hpp

```diff
@@ -0,0 +1,171 @@
+        std::chrono::time_point<std::chrono::system_clock> last_run;
+        /// \brief Only run the pre function once.
+        bool run_pre_once_only = true;
+        /// \brief Run post only once.
+        bool run_post_once_only  = true;
+        /// \brief User defined data to be handled by the task.
+        std::any data;
+    public:
+        ///  \brief Returns true if the task is ready to run
+        /// \retval true if sufficient time has elapsed
+        bool ready() const {
+            auto now = std::chrono::system_clock::now();
+            return (now - last_run) > interval;
+        }
+        /**
+         * \brief Returns true if the task has run before, as determined by the last run timestamp
+         *
+         * \retval true if the task has run before
+         */
+        bool has_run()const{
+            return last_run.time_since_epoch().count() == 0;
+        }
+        /**
+         * \brief Run the task
+         */
+        void run(){
+            if( !ready() ){
+                return;
+            }
+            if( (run_pre_once_only && !has_run()) || !run_pre_once_only){
+                pre(data);
+            }
+            execute(data);
+            if( (run_post_once_only && !has_run() ) || !run_post_once_only){
+                post(data);
+            }
+            last_run=std::chrono::system_clock::now();
+        }
+    }; // struct cron_task
+
+    /// \brief Builder for a CronTask.
+    struct cron_builder {
+        std::function<void(std::any&)> to_execute_    = detail::noop;
+        std::function<void(std::any&)> pre_function_  = detail::noop;
+        std::function<void(std::any&)> post_function_ = detail::noop;
+        bool run_pre_once_ = true;
+        bool run_post_once_ = true;
+        int interval_ = 0;
+        std::any data_;
+
+        /// \brief set the pre action to be run before the main function
+        ///
+        /// \param[in] f The function to run before the main event, whatever that might be
+        cron_builder& pre(std::function<void(std::any&)> f){pre_function_ = f; return *this;}
+
+        /// \brief set the post function, to be executed before the execution function
+        ///
+        /// \param[in] f The function to be run after the primary function of the task. By default only run once
+        cron_builder& post(std::function<void(std::any&)> f){post_function_ = f;return *this;}
+
+        /// \brief Set the execution function to be run.
+        ///
+        /// \param[in] The main function to be executed
+
+        cron_builder& to_execute(std::function<void(std::any&)> f){to_execute_ = f;return *this;}
+
+        /// \brief Set the interval in seconds between runs of this task. A value of Zero(the default), results
+        /// in the task being run every time the cron manager loops.
+        ///
+        /// \param[in] n The number of seconds between runs of the task
+        cron_builder& interval(int n){interval_ = n; return *this;}
```

Comment:
maybe use `std::chrono::seconds` as the parameter type to express intent.

this is mainly for the reader of your class.